pkg/webhook/policy.go

```diff
@@ -501,14 +508,83 @@ func (h *validationHandler) reviewRequest(ctx context.Context, req *admission.Re
 		}
 	}
 
+	review, err := h.createReviewForRequest(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create augmentedReview: %s", err)
+	}
+
+	// Convert the request's generator resource to unstructured for expansion
+	obj := &unstructured.Unstructured{}
+	if _, _, err := deserializer.Decode(req.Object.Raw, nil, obj); err != nil {
```

Comment:
Deserialization is computationally expensive and we're currently doing it in multiple places, see:

https://github.com/open-policy-agent/gatekeeper/blob/841e10000acf3e7fc42e35a0ab1be5dfbe0138ae/pkg/target/matcher.go#L67-L87

We should probably optimize that at some point (not a blocker for this PR)

Reply:
created https://github.com/open-policy-agent/gatekeeper/issues/2195 to track